### PR TITLE
Migrate bicep example: 201/vmss-windows-autoscale

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/201/vmss-windows-autoscale

The corresponding PR for the modified bicep example is here: https://github.com/Azure/bicep/pull/3342

### CODE REVIEW

File: quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+          {
+            name: nicName
+            properties: {
+              primary: true
+              ipConfigurations: [
+                {
+                  name: ipConfigName
+                  properties: {
+                    subnet: {
+                      id: '${virtualNetwork.id}/subnets/${subnetName}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep:162]: seems like we should be consistent with the construction of resourceIds that bicep doesn't handle for us... either use the resourceId() function or string interp...

resourceId() probably makes the most sense, consistent, familiar and intentional.

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep (hunk 1 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep:162]: Or better, maybe the subnet a resource so we can call subnet.id

Comment (bmoore-msft) [reply quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep:162]: that works too - though more of a refactor and causes downtime on redeploy.  We also have other resources that don't have children but still have resourceIds.

File: quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnetName)
+                    }
+                    loadBalancerBackendAddressPools: [
+                      {
+                        id: resourceId('Microsoft.Network/loadBalancers/backendAddressPools', loadBalancer.name, bePoolName)
+                      }
+                    ]
+                    loadBalancerInboundNatPools: [
+                      {
+                        id: '${loadBalancer.id}/inboundNatPools/${natPoolName}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep:171]: use resourceId here as above

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep (hunk 2 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep:171]: inboundNatPools doesn't seem to be a resource, so I don't think I can use resourceId(), left as is

Comment (bmoore-msft) [reply quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep:171]: Welcome to NRP - it is a resourceId - see: https://docs.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-sets/create-or-update#virtualmachinescalesetipconfiguration or you can peek at the other samples in the repo

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by StephenWeatherford** [comment]
"id": "[resourceId('Microsoft.Network/loadBalancers/inboundNatPools', variables('loadBalancerName'), variables('natPoolName'))]"

**Pull request merged**